src/aosm/azext_aosm/generate_nsd/nf_ret.py

```diff
@@ -0,0 +1,165 @@
+# --------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT
+# License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------
+"""Handles the creation of a resource element template for a network function."""
+
+import json
+
+from typing import Dict, Any, List, Union
+from knack.log import get_logger
+
+from azext_aosm._configuration import NFDRETConfiguration
+from azext_aosm.util.constants import CNF, VNF
+from azext_aosm.util.management_clients import ApiClients
+from azext_aosm.vendored_sdks.models import NetworkFunctionDefinitionVersion, NFVIType
+
+
+logger = get_logger(__name__)
+
+
+class nfRET:
+    """
+    Represents a single network function resource element template withing an NSD.
+    """
+
+    def __init__(
+        self, api_clients: ApiClients, config: NFDRETConfiguration, cg_schema_name: str
+    ) -> None:
+        self.config = config
+        self.cg_schema_name = cg_schema_name
+        nfdv = self._get_nfdv(config, api_clients)
+        print(
+            f"Finding the deploy parameters for {self.config.name}:{self.config.version}"
+        )
+
+        if not nfdv.deploy_parameters:
+            raise NotImplementedError(
+                f"NFDV {self.config.name} has no deploy parameters, cannot generate NSD."
+            )
+        self.deploy_parameters: Dict[str, Any] = json.loads(nfdv.deploy_parameters)
+
+        self.nf_type = self.config.name.replace("-", "_")
```

Comment:
Confusing. I would have thought NF type would be CNF or VNF based on other code. Rename to something else?